src/plugins/silentMessageToggle.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Vencord, a modification for Discord's desktop app
+ * Copyright (c) 2023 Vendicated and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { addPreSendListener, removePreSendListener, SendListener } from "@api/MessageEvents";
+import ErrorBoundary from "@components/ErrorBoundary";
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { Button, ButtonLooks, ButtonWrapperClasses, React, Tooltip } from "@webpack/common";
+
+function SilentMessageToggle() {
+    const [enabled, setEnabled] = React.useState(false);
+
+    React.useEffect(() => {
+        const listener: SendListener = (_, message) => {
+            if (enabled) {
+                setEnabled(false);
+                if (!message.content.startsWith("@silent ")) message.content = "@silent " + message.content;
+            }
+        };
+
+        addPreSendListener(listener);
+        return () => void removePreSendListener(listener);
+    }, [enabled]);
+
+    return (
+        <Tooltip text="Toggle Silent Message">
+            {tooltipProps => (
+                <Button
+                    {...tooltipProps}
+                    onClick={() => setEnabled(!enabled)}
```

Comment:
```suggestion
                    onClick={() => setEnabled(prev => !prev)}
```

when using the old state you should always use the function update style instead of using the variable directly as it's not guaranteed the variable is up to date

Reply:
could also useReducer here
```ts
const [enabled, toggleEnabled] = React.useReducer(x => !x, false);
```